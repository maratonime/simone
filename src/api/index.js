// @flow
import { getContests } from './getContests';

export default {
  getContests,
};
