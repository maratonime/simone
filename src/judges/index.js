// @flow
import codeforces from './codeforces';
import csacademy from './csacademy';
import codechef from './codechef';

export default {
  codeforces: codeforces,
  csacademy: csacademy,
  codechef: codechef,
};
