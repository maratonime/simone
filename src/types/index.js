// @flow

export type { Contest } from './contest';
export type { Judge, State } from './contest';
