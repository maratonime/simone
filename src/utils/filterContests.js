// @flow
import type { Contest } from '../types';

export function filterContests(
  contests: Array<Contest>,
  from?: Date,
  to?: Date,
): Array<Contest> {
  contests = contests.filter(contest => {
    if (from instanceof Date && contest.startTime instanceof Date)
      return contest.startTime >= from;
    else return true;
  });
  contests = contests.filter(contest => {
    if (to instanceof Date && contest.startTime instanceof Date)
      return contest.startTime <= to;
    else return true;
  });
  return contests;
}
