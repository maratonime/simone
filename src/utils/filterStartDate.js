// @flow

import type { Contest } from '../types';

export function filterStartDate(
  contests: Array<Contest>,
  from?: Date,
  to?: Date,
): Array<Contest> {
  contests = contests.filter(contest => {
    if (from != null)
      if (contest.startTime != null) return contest.startTime >= from;
      else return false;
    else return true;
  });
  contests = contests.filter(contest => {
    if (to != null)
      if (contest.startTime != null) return contest.startTime <= to;
      else return false;
    else return true;
  });
  return contests;
}
