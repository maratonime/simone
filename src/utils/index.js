// @flow

export { filterStartDate } from './filterStartDate';
