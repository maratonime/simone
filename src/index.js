// @flow

export { default as judges } from './judges';
import api from './api';
export const { getContests } = api;
