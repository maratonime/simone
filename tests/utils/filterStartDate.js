// @flow

import { filterStartDate } from '../../src/utils';

test('filter contests', () => {
  const cs = [];
  for (let i = 1; i <= 10; i++)
    cs.push({
      code: '1',
      name: 'name',
      judge: 'codeforces',
      url: 'https://www.youtube.com/watch?v=dQw4w9WgXcQ',
      startTime: new Date(i * 1000),
    });
  expect(filterStartDate(cs, new Date(1000), new Date(7000))).toMatchSnapshot();
});
