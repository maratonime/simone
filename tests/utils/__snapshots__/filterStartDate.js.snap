// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`filter contests 1`] = `
Array [
  Object {
    "code": "1",
    "judge": "codeforces",
    "name": "name",
    "startTime": 1970-01-01T00:00:01.000Z,
    "url": "https://www.youtube.com/watch?v=dQw4w9WgXcQ",
  },
  Object {
    "code": "1",
    "judge": "codeforces",
    "name": "name",
    "startTime": 1970-01-01T00:00:02.000Z,
    "url": "https://www.youtube.com/watch?v=dQw4w9WgXcQ",
  },
  Object {
    "code": "1",
    "judge": "codeforces",
    "name": "name",
    "startTime": 1970-01-01T00:00:03.000Z,
    "url": "https://www.youtube.com/watch?v=dQw4w9WgXcQ",
  },
  Object {
    "code": "1",
    "judge": "codeforces",
    "name": "name",
    "startTime": 1970-01-01T00:00:04.000Z,
    "url": "https://www.youtube.com/watch?v=dQw4w9WgXcQ",
  },
  Object {
    "code": "1",
    "judge": "codeforces",
    "name": "name",
    "startTime": 1970-01-01T00:00:05.000Z,
    "url": "https://www.youtube.com/watch?v=dQw4w9WgXcQ",
  },
  Object {
    "code": "1",
    "judge": "codeforces",
    "name": "name",
    "startTime": 1970-01-01T00:00:06.000Z,
    "url": "https://www.youtube.com/watch?v=dQw4w9WgXcQ",
  },
  Object {
    "code": "1",
    "judge": "codeforces",
    "name": "name",
    "startTime": 1970-01-01T00:00:07.000Z,
    "url": "https://www.youtube.com/watch?v=dQw4w9WgXcQ",
  },
]
`;
